pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -0,0 +1,316 @@
+        CI will begin falling being nightly CoreOS content.
+        """
+
+        # Only for applicable versions
+        major, minor = [int(n) for n in self.version.split('.')]
+        if major <= 4 and minor < 12:
+            return
+
+        # Only for stream assembly and openshift-{MAJOR}.{MINOR} branches
+        if not self.assembly == 'stream' or self.doozer_data_gitref:
+            return
+
+        try:
+            # OC registry login
+            await exectools.cmd_gather_async(
+                f'oc --kubeconfig {os.environ["KUBECONFIG"]} registry login')
+
+            supported_arches = await branch_arches(f'openshift-{self.version}')
+            tags_to_transfer: list = await self._tags_to_transfer()
+
+            tasks = []
+            for arch in supported_arches:
+                arch_suffix = GO_ARCH_SUFFIXES[arch]
+                for tag in tags_to_transfer:
+                    tasks.append(self._tag_into_ci_imagestream(arch_suffix, tag))
+            await asyncio.gather(*tasks)
+
+        except (ChildProcessError, KeyError) as e:
+            slack_client = self.runtime.new_slack_client()
+            slack_client.bind_channel(f'openshift-{self.version}')
+            await slack_client.say(
+                f'Unable to mirror CoreOS images to CI for {self.version}: {e}')
+
+    async def _update_nightly_imagestreams(self):
+        """
+        Determine the content to update in the ART latest imagestreams and apply those changes on the CI cluster.
+        The verb will also mirror out images to the quay monorepo.
+        """
+
+        self.logger.info('Generating and applying imagestream updates')
+        mirror_working = 'MIRROR_working'
+
+        # Run release:gen-payload
+        cmd = [
+            'doozer',
+            f'--assembly={self.assembly}']
+        if self.images:
+            cmd.append(f'--images={self.images}')
+        cmd.extend([
+            f'--working-dir={mirror_working}',
+            f'--data-path={self.data_path}',
+            f'--group=openshift-{self.version}',
+            'release:gen-payload',
+            f'--output-dir={GEN_PAYLOAD_ARTIFACTS_OUT_DIR}',
+            '--apply'
+        ])
+        if self.emergency_ignore_issues:
+            cmd.append('--emergency-ignore-issues')
+        if not self.skip_multiarch_payload:
+            cmd.append('--apply-multi-arch')
+        if self.exclude_arches:
+            cmd.extend([f'--exclude-arch {arch}' for arch in self.exclude_arches])
+        if self.runtime.dry_run:
+            cmd.extend(['--skip-gc-tagging', '--moist-run'])
+        await exectools.cmd_assert_async(cmd)
+
+        # Populate CI imagestreams
+        await self._populate_ci_imagestreams()
+
+        if self.publish:
+            # OC registry login
+            await exectools.cmd_gather_async(
+                f'oc --kubeconfig {os.environ["KUBECONFIG"]} registry login')
```

Comment:
This will require `KUBECONFIG` is present.
We can just passthrough all env vars to the subprocess:
```suggestion
            await exectools.cmd_gather_async(
                f'oc registry login', env=os.environ.copy())
```